src/rgw/rgw_lc.cc

```diff
@@ -43,6 +55,49 @@ void RGWLifecycleConfiguration::_add_rule(LCRule *rule)
   prefix_map[rule->get_prefix()] = rule->get_expiration().get_days();
 }
 
+bool RGWLifecycleConfiguration::check_and_add_rule(LCRule *rule)
+{
+  if (!rule->validate())
+    return false;
+  string id;
+  rule->get_id(id);
+  if (rule_map.find(id) != rule_map.end())  //id shouldn't be the same 
+    return false;
+  rule_map.insert(pair<string, LCRule>(id, *rule));
+
+  auto ret = prefix_map.insert(pair<string, int>(rule->get_prefix(), rule->get_expiration().get_days()));
+  //Now prefix shouldn't be the same. When we add noncurrent expiration or other action, prefix may be same.
+  if (!ret.second)
+    return false;
+  return true;
+}
+
+//Rules are conflicted: if one rule's prefix starts with other rule's prefix, and these two rules
+//define same action(now only support expiration days). 
+bool RGWLifecycleConfiguration::validate() 
```

Comment:
Yes, I just compare adjacent entries. The prefix_map's type is 'map<string, int>', so the elements in it are sorted by the rule's prefix. Assumes that "a"<A<"abc", prefix "abc" begins with prefix "a", then A must start with "a". So we can just compare adjacent entries. At present, if there are two rules like above, we can say lifecycle config is illegal because of only expire day is supported. When we have other action, we need to compare three rules like above(or four and so on), but not a full comparison. What do you think about this?